Source/JNA/waffle-jna/src/main/java/waffle/servlet/WindowsPrincipal.java

```diff
@@ -255,4 +255,30 @@ public String toString() {
         return this.getName();
     }
 
+    /*
+     * (non-Javadoc)
+     * @see java.lang.Object#equals(java.lang.Object)
+     */
+    @Override
+    public boolean equals(final Object o) {
```

Comment:
Well, what I meant is that [inheritance should be forbidden in general on a class that implements equals](http://stackoverflow.com/questions/12239344/why-should-i-not-use-equals-with-inheritance). One potential problem though is that it can break backwards compatibility for people who mock this class, but I can argue that data types should be stubbed rather than mocked.